### DISCUSSION TITLE
How to run 13B GPTQ models with groupsize 32 (best quality) on the card with 12Gb VRAM.

### DISCUSSION THREAD

**Discussion by valden80** [oobabooga/text-generation-webui#4125] [Show and tell]
TheBloke usually has several version of the every model in his archive, the best ones (in term of output quality) from 4 bit GPTQ is version with groupsize 32.
There is only one problem. If you own card like RTX 3060 with 12GB of VRAM, and still want to use 4096 of context - you will get OOM when trying to use this models "as is" in webui. You can set smaller context (about 2600), but this is lower quality of your experience in chats ("goldfish memory" problem with characters).

I found how to use these models on 3060 12GB with full context. Only on linux, though with special setup.

1. You need to setup you system in such way, that 3060 is not used for the graphic at all. Even for the desktop. Obviously, your system should use another GPU for it (integrated from CPU, or second card), or run headless without GUI (X/Wayland). This will leave all 12GB of VRAM free for the model.
2. You need to start webui with this line: 
  `CUDA_MODULE_LOADING=LAZY PYTORCH_CUDA_ALLOC_CONF=garbage_collection_threshold:0.8,max_split_size_mb:24 ./start_linux.sh`
3. Load model through exllama or exllama_hf

This way typical 13B model with groupsize 32 take ~11000кб of VRAM after loading, and ~11850-11950Kb at peaks in the generation process. 
Extremely tight, but still working without OOM's. 
IMHO - it's worth the trouble to setup, as I can see the difference in quality of generation between gs 128 ("main" branches in TheBlocke archives) and gs 32 by my own eyes.
Speed of generation with this setup is slightly slower, but still pretty good. On my 3060 it's around 10-12 t/s.

I hope this will be useful for someone.

**Comment by maddog7667** [comment]
Interesting, I too noticed a difference between gs 128 and gs 32. It reminds me of mp3 compression when it first came out for audio. gs 128 is like 128kbs mp3 vs 32gs is like 320kbs. You notice the difference.

**Comment by nbollman** [comment]
Before I knew better, Id download default model with no groupsize, Ive been only downloading GPTQ 4bit 32gs models for awhile now, they're minimally slower and only slightly bigger in vram and between no groupsize and 32gs there is a difference.
For the love of exllamav2 WSL/Linux is the only way to go, Linux will definetly give you more system ram...
